File: Day1_PokemonStatsApp/PokemonStatsApp/Data Models/StatInfo.swift
//
//  StatInfo.swift
//  PokemonStatsApp
//
//  Created by Sanket Sonje on 15/01/25.
//

import Foundation

/// Contains detailed information about a specific stat type
struct StatInfo: Codable {

    /// The name of the stat (e.g., "hp", "attack", "defense")
    let name: String

    /// URL containing additional information about the stat
    let url: String
}

File: Android_PokemonApp/app/src/main/java/com/example/android_pokemonapp/data/model/StatInfo.kt
package com.example.android_pokemonapp.data.model

import com.google.gson.annotations.SerializedName

/**
 * Represents a type of Pokemon statistic.
 *
 * Defines a base statistical attribute that all Pokemon possess,
 * such as HP, Attack, Defense, Special Attack, Special Defense, or Speed.
 *
 * @property name The name of the statistic (e.g., "hp", "attack", "defense")
 * @property url API endpoint URL for detailed information about this statistic
 * @see Stat
 */
data class StatInfo(
	@SerializedName("name")
	val name: String,

	@SerializedName("url")
	val url: String
)
